lib/matplotlib/dviread.py

```diff
@@ -925,48 +979,43 @@ def __iter__(self):
     def _parse(self, file):
         result = []
 
-        state = 0
-        for line in file:
-            comment_start = line.find('%')
-            if comment_start > -1:
-                line = line[:comment_start]
-            line = line.strip()
+        lines = (line[:line.find(b'%')] if b'%' in line else line.strip()
```

Comment:
The previous code ran `strip` after removing the comment as well; this can be shortened a bit: `line.split(b'%')[0].strip()`.